spec/support/xml_matcher.rb

```diff
@@ -0,0 +1,18 @@
+RSpec::Matchers.define :match_xml do |comparison|
+  diffable
+
+  # REXML::Documents do not implement comparison; they are never ==.
+  # This matcher considers the documents the same if their string outputs
+  # are equal after both going through the REXML::Formatters::Pretty.
+  match do |document|
+    # We have to override these for the diff to use these, rather than input strings
```

Comment:
The `diffable` line would otherwise print the diff between `document` and `comparison`, which is the _original_ XML before normalization. This cause some pulling out of hair.